### PR TITLE
New + Fix : fixed some errors on showLinkToObjectBlock function and added option to link propales to interventions

### PR DESCRIPTION
**Pull request by bafbes**

### CODE REVIEW

File: htdocs/comm/propal.php [hunk 1]
```diff
@@ -2252,10 +2252,16 @@
 		// Linked object block
 		$somethingshown = $form->showLinkedObjectBlock($object);
 
-		// Show links to link elements
-		$linktoelem = $form->showLinkToObjectBlock($object);
-		if ($linktoelem) print '<br>'.$linktoelem;
```
Comment (eldy) [inline htdocs/comm/propal.php:2257]: The goal of this function is to show ALL linked objects, whatever is the type.
So why intoducing specific code for intervention ? If intervention is linked to proposal, it should appears with this generic code with no need of change (if not, pb is inside function showLinkToObjectBlock)

Comment (bafbes) [inline htdocs/comm/propal.php:2257]: As you said, the goal of showLinkedObjectBlock is  to show ALL linked objects, whatever is the type. It hasn't been touched.

The goal of showLinkToObjectBlock function is to show links to link the current object to other object types.

This function previously contained only implementation of links to order objects. It contains now the implementation of two other elements ( interventions and supplier orders), and the list is subject to be extended to contain all common objects in the future.

As it makes no sense to link some element types to other ones (ex : customer orders to  expense report) I proposed the solution to use the array $restrictlinksto, and include in it only the elements allowed to be linked to the current object, and I started by providing the possibility to link propales to only interventions based on the value of PROPALE_LINK_TO_INTERVENTION.  The default behaviour would show links to all types defined in showLinkToObjectBlock.

I think the matter is subject to discussion !

**Comment by eldy** [comment]
You're right when you say we must be able to link to other objects, not only orders. Yes we should be able to link an intervention on a proposal and vice versa manually. But I would like to address this in a more generic manner than using a block of code for each type of object. I will implement something to show the way soon...

**Comment by bafbes** [comment]
I continue to follow the development of this issue

**Comment by eldy** [comment]
I pushed a new generic code to allow to link elements with others.
Take a look and see if it can address your need.

**Pull request closed without merging**